lib/nolibc/include/stdlib.h

```diff
@@ -61,60 +61,16 @@ unsigned long long strtoull(const char *nptr, char **endptr, int base);
  */
 int atoi(const char *s);
 long atol(const char *s);
-
 #if CONFIG_LIBUKALLOC
-/* Allocate size bytes of memory. Returns pointer to start of allocated memory,
- * or NULL on failure.
- */
-static inline void *malloc(size_t size)
-{
-	return uk_malloc(uk_alloc_get_default(), size);
-}
-
-/* Release memory previously allocated by malloc(). ptr must be a pointer
- * previously returned by malloc(), otherwise undefined behavior will occur.
- */
-static inline void free(void *ptr)
-{
-	return uk_free(uk_alloc_get_default(), ptr);
-}
-
-/* Allocate memory for an array of nmemb elements of size bytes. Returns
- * pointer to start of allocated memory, or NULL on failure.
- */
-static inline void *calloc(size_t nmemb, size_t size)
-{
-	return uk_calloc(uk_alloc_get_default(), nmemb, size);
-}
 
-/* Change the size of the memory block pointed to by ptr to size bytes.
- * Returns a pointer to the resized memory area. If the area pointed to was
- * moved, a free(ptr) is done.
- */
-static inline void *realloc(void *ptr, size_t size)
-{
-	return uk_realloc(uk_alloc_get_default(), ptr, size);
-}
-
-/* Allocate size bytes of memory, aligned to align bytes, and return the
- * pointer to that area in *memptr. Returns 0 on success, and a non-zero error
- * value on failure.
- */
-static inline int posix_memalign(void **memptr, size_t align, size_t size)
-{
-	return uk_posix_memalign(uk_alloc_get_default(),
-				 memptr, align, size);
-}
-
-/* Allocate size bytes of memory, aligned to align bytes. Returns pointer to
- * start of allocated memory, or NULL on failure.
- */
-static inline void *memalign(size_t align, size_t size)
-{
-	return uk_memalign(uk_alloc_get_default(), align, size);
-}
-#endif /* CONFIG_LIBUKALLOC */
+void *malloc(size_t size);
```

Comment:
I suggest that we keep the comments regarding functions in the header file, not in the source file. 

@skuenzer @razvand any opinions here?

Reply:
I was uncertain about the coding standard in this case, as I believed the existing comments provided more detailed descriptions of the functions' operations. This seemed different from the Doxygen style comments, which, from what I understand, are typically placed in the header files. I'll wait for the other to tune in and give me the green-light regarding the decision.

Reply:
I agree with @Starnox' s point. There must be a Doxygen-style comment in the header file, that documents the exposed API of the function, i.e. information that is related to the signature of the function (return value, parameters, parameter types).

However, internal details on the implementation of the function, i.e. its body, go to the source code. These either go inside the function body itself (inline comments), or on top of the function body + signature. These comments go in the source code because they would detail internal information pertaining to the implementation (internal variables, other functions being called, design / implementation decisions) that don't make sense to be in the header file, that doesn't provide any glimpse into the implementation.